sklearn/manifold/isomap.py

```diff
@@ -54,9 +54,13 @@ class Isomap(BaseEstimator, TransformerMixin):
 
         'D' : Dijkstra's algorithm.
 
-    neighbors_algorithm : string ['auto'|'brute'|'kd_tree'|'ball_tree']
+    neighbors_algorithm : string ['auto'|'brute'|'kd_tree'|'ball_tree'| \
+            'precomputed']
         Algorithm to use for nearest neighbors search,
         passed to neighbors.NearestNeighbors instance.
+        If "precomputed", X is assumed to be a distance matrix and
+        must be square. X may be a sparse matrix, in which case only "nonzero"
```

Comment:
I've been wondering whether we should change our terminology to refer to "sparse graphs" and "connected" instead of "sparse matrix" and "nonzero"... not sure which helps users more